File: CryptoFront/src/app/models/comment.ts
import { ReactionCount } from "./reaction-count";
import { Reaction } from "src/app/models/reaction";
import { User } from "src/app/models/user";

export class Comment {
    public user: User;
    public children: Comment[];
    public reactions: Reaction[];
    public reactionCounts: ReactionCount[];
    public text: string;
}

File: CryptoBack/Models/Comment.cs
using CryptoBack.Models.Abstract;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace CryptoBack.Models
{
    public class Comment : VersionedEntity, IComposite<Comment>
    {
        [ForeignKey("User.Id")]
        public long UserId { get; set; }

        [ForeignKey("Article.Id")]
        public long? ArticleId { get; set; }

        [ForeignKey("Comment.Id")]
        public long? CommentId { get; set; }


        public virtual User User { get; set; }

        public virtual Article Article { get; set; }

        public virtual Comment Parent { get; set; }

        public virtual IList<Comment> Children { get; set; }

        public virtual IList<Reaction> Reactions { get; set; }


        public string Text { get; set; }


        [NotMapped]
        public IList<ReactionCount> ReactionCounts { get; set; }
    }
}
